pkg/envtest/server.go

```diff
@@ -62,14 +62,17 @@ const (
 	defaultKubebuilderControlPlaneStopTimeout  = 20 * time.Second
 )
 
-// Default binary path for test framework
-func defaultAssetPath(binary string) string {
-	assetPath := os.Getenv(envKubebuilderPath)
-	if assetPath == "" {
-		assetPath = defaultKubebuilderPath
+// getBinAssetPath will return the path for the binary informed
+func (te *Environment) getBinAssetPath(binary string) string {
+	if hasBinary(binary, te.BinaryDirectoryPath) {
```

Comment:
The `hasBinary` pattern doesn't work here, because we want to inform the envtest caller that the binary does not exist at the path they specified and not at the default path, which is what this method currently does. Instead, just check if each data variable is set, and if so use it as the parent dir.
```suggestion
	if te.BinaryDirectoryPath != "" {
```

Reply:
I'd thought that would be nice if envtest was able to check if the bin exists or not and then try to check the follow-up the options according to the priority. However, I can agree that it is out of the scope over `Allow passing the assets path bin via the env config` so I changed it as you suggested. 👍